### ISSUE TITLE
belongs_to association not loaded with rails 4.2.1

### ISSUE THREAD

**Issue by mfazekas** [makandra/active_type#28]
```ruby
unless File.exist?('Gemfile')
  File.write('Gemfile', <<-GEMFILE)
    source 'https://rubygems.org'
    gem 'rails', 4.2.1.rc2
    gem 'sqlite3'
    gem 'active_type', '0.3.3'
  GEMFILE

  system 'bundle'
end
require 'bundler'
Bundler.setup(:default)
require 'active_type'
require 'active_record'
require 'minitest/autorun'
require 'logger'
Minitest::Test = MiniTest::Unit::TestCase unless defined?(Minitest::Test)
ActiveRecord::Base.establish_connection(adapter: 'sqlite3', database: ':memory:')
ActiveRecord::Base.logger = Logger.new(STDOUT)

ActiveRecord::Schema.define do
  create_table :posts, force: true do |t|
  end

  create_table :comments, force: true do |t|
    t.integer :post_id
  end
end

class Post < ActiveRecord::Base
end

class CommentObj < ActiveType::Object
  belongs_to :post
  attribute :post_id, :integer
end

class Comment < ActiveRecord::Base
  belongs_to :post
end

class BugTest < Minitest::Test
  def test_association_stuff
    post = Post.create!
    comment_obj = CommentObj.new post_id:post.id
    comment = Comment.new post_id:post.id
   
    assert_equal post, comment.post
    assert_equal post, comment_obj.post
  end
end
```

The line ```assert_equal post, comment_obj.post``` fails. Seems to work with 4.2.0.

**Comment by mfazekas** [comment]
Can someone take a look? I've just tested 4.2.1rc2 and the issue is there, so it would be nice to have it fixed before rails 4.2.1 goes out.

**Comment by henning-koch** [comment]
Released as 0.3.4.
